fastlane_core/lib/fastlane_core/helper.rb

```diff
@@ -151,6 +151,32 @@ def self.transporter_path
       return File.join(self.itms_path, 'bin', 'iTMSTransporter')
     end
 
+    def self.keychain_path(name)
+      # Existing code expects that a keychain name will be expanded into a default path to Libary/Keychains
+      # in the user's home directory. However, this will not allow the user to pass an absolute path
+      # for the keychain value
+      #
+      # So, if the passed value can't be resolved as a file in Library/Keychains, just use it as-is
+      # as the keychain path.
+      #
+      # We need to expand each path because File.exist? won't handle directories including ~ properly
+      #
+      # We also try to append `-db` at the end of the file path, as with Sierra the default Keychain name
+      # has changed for some users: https://github.com/fastlane/fastlane/issues/5649
+      #
+
+      keychain_paths = [
```

Comment:
It might be worthwhile to append `.keychain` to the name here if it doesn't already include it, what do you think?